demo/src/app/components/tabset/tabset.component.ts

```diff
@@ -1,23 +1,31 @@
 import {Component} from '@angular/core';
+
 import {ContentWrapper} from '../../shared';
 import {ExampleBoxComponent, NgbdApiDocs} from '../shared';
-import {DEMO_DIRECTIVES, DEMO_SNIPPETS} from './demos';
+import {TabsetBasicComponent, basicHtmlContent, basicTsContent,
+        TabsetPillsComponent, pillsHtmlContent, pillsTsContent,
+        TabsetSelectByIdComponent, selectByIdHtmlContent, selectByIdTsContent,
+        TabsetPreventChangeComponent, preventChangeHtmlContent, preventChangeTsContent} from './demos';
 
 @Component({
   selector: 'ngbd-tabs',
-  template: `
-    <ngbd-content-wrapper component="Tabs">
-      <ngbd-api-docs directive="NgbTabset"></ngbd-api-docs>
-      <ngbd-api-docs directive="NgbTab"></ngbd-api-docs>
-      <ngbd-api-docs directive="NgbTabTitle"></ngbd-api-docs>
-      <ngbd-api-docs directive="NgbTabContent"></ngbd-api-docs>
-      <ngbd-example-box demoTitle="Tabset" [htmlSnippet]="snippets.basic.markup" [tsSnippet]="snippets.basic.code">
-        <ngbd-tabset-basic></ngbd-tabset-basic>      
-      </ngbd-example-box>
-    </ngbd-content-wrapper>
-  `,
-  directives: [ContentWrapper, NgbdApiDocs, DEMO_DIRECTIVES, ExampleBoxComponent]
+  template: require('./tabset.component.html'),
+  directives: [
+    ContentWrapper,
+    NgbdApiDocs,
+    TabsetBasicComponent,
+    TabsetPillsComponent,
+    TabsetSelectByIdComponent,
+    TabsetPreventChangeComponent,
+    ExampleBoxComponent]
 })
 export class NgbdTabs {
-   snippets = DEMO_SNIPPETS;
+  basicHtmlContent = basicHtmlContent;
+  basicTsContent = basicTsContent;
+  pillsHtmlContent = pillsHtmlContent;
+  pillsTsContent = pillsTsContent;
+  selectByIdHtmlContent = selectByIdHtmlContent;
+  selectByIdTsContent = selectByIdTsContent;
+  preventChangeHtmlContent = preventChangeHtmlContent;
+  preventChangeTsContent = preventChangeTsContent;
```

Comment:
You shouldn't be adding those lines - everything should be part of `DEMO_SNIPPETS` already